docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -146,12 +231,154 @@ Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoke
 
 Note that methods annotated with `@QueryMethod` can be invoked even when a Workflow is in the `Completed` state.
 
-### Workflow implementation constraints
+### Typed and Untyped `WorkflowStubs`
+
+A `WorkflowStub` is a proxy of your Workflow implementation. To start a Workflow Execution, we need to create a `WorkflowStub`.
+There are two types of `WorkflowStubs`: Typed and Untyped.
+
+**Typed `WorkflowStubs`**
+
+A typed `WorkflowStub` returns an implementation of your Workflow interface. For example consider the following Workflow interface:
+
+```java
+@WorkflowInterface
+public interface NotifyUserAccounts {
+    @WorkflowMethod
+    void notifyAccount(String[] accountIds);
+
+    @QueryMethod
+    int getCount();
+}
+```
+
+The Workflow implementation for the preceding Workflow interface using typed `WorkflowStub` will be as follows:
+
+```java
+Public static void main(String[] args){
+NotifyUserAccounts workflow = client.newWorkflowStub(
+      NotifyUserAccounts.class,
+      WorkflowOptions.newBuilder()
+            .setWorkflowId("notifyAccounts")
+            .setTaskQueue(taskQueue)
+            .build()
+        );
+}
+```
+
+Typed `WorkflowStub` are useful because they are type safe. You can invoke your Workflow methods such as [`@WorkflowMethod`](# `@WorkflowMethod`), [`QueryMethod`](# `QueryMethod` ), and [`@SignalMethod`](# `QueryMethod`) directly.
+
+**Untyped `WorkflowStub`**
+
+An untyped `WorkflowStub` does not use the Workflow interface, and is not typesafe. It is more flexible because it has methods from the `WorkflowStub` interface, such as `start`, `signalWithStart`, `getResults` (sync and async), `query`, `signal`, `cancel` and `terminate`.
+Note that the Temporal Java SDK also provides typed `WorkflowStub` versions for these methods.
+
+See the `WorkflowStub.java` reference: <https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java>
+
+When using untyped `WorkflowStubs`, we rely on the Workflow Type, Activity Type, Child Workflow Type, as well as Query and Signal names.
+
+See [How to spawn a Workflow Execution in Java](/docs/java/how-to-spawn-a-workflow-execution-in-java).
+
+### Calling other Workflows
+
+Workflows can invoke, and send Signals to, other Workflows purely by name. This helps particularly for executing Workflows written in other language SDKs.
```

Comment:
does this sentence need the second comma after "to"?

Reply:
Yes, to close the appositive "and send Signals to". (The sentence remains grammatically correct without the appositive.) The alternative is to remove both commas.

Reply:
leaving the commas in because I want to separate the ideas of invoking workflows and sending Signals to Workflows.